src/Libraries/CoreNodeModels/Input/BaseTypes.cs

```diff
@@ -135,7 +135,7 @@ public override NodeInputData InputData
                     Name = this.Name,
                     Type = NodeInputTypes.numberInput,
                     Description = this.Description,
-                    Value = System.Convert.ToDouble(Value),
+                    Value = Value,
```

Comment:
yes - `NumberInputNode` ... `doubleInput`'s value is string for some reason. Probably because you type into this node and it stores the entered string I guess.